fire/core.py

```diff
@@ -241,12 +241,17 @@ def _IsHelpShortcut(component_trace, remaining_args):
   return show_help
 
 
-def _PrintResult(component_trace, verbose=False):
+def _PrintResult(component_trace, verbose=False, serialize=None):
   """Prints the result of the Fire call to stdout in a human readable way."""
   # TODO(dbieber): Design human readable deserializable serialization method
   # and move serialization to its own module.
   result = component_trace.GetResult()
 
+  # Allow users to modify the return value of the component and provide 
+  # custom formatting.
+  if callable(serialize):
+    result = serialize(result)
```

Comment:
We'll want to leave result unchanged, and introduce a new variable serialized_result that we use from here on out.
This way in interactive mode (-- -i) the user will still have access to the non-serialized result, as well as serialized_result.

Reply:
This is internally scoped inside `_PrintResult` and doesn't actually affect the value of the interactive result, unless I'm missing some weird introspection thing that's happening

Reply:
My bad. You're right, it's fine as is.

(The relevant place is `variables['result'] = component` at line 611, which as you say isn't affected by what's happening here.)